legal-api/src/legal_api/services/involuntary_dissolution.py

```diff
@@ -45,18 +45,26 @@ class EligibilityDetails:
         ar_overdue: bool
         transition_overdue: bool
 
+    @dataclass
+    class ExclusionSettings:
+        """Details about the exclude of a business for dissolution."""
+
+        exclude_in_dissolution: bool = True
+        exclude_future_effective_filing: bool = False
+
     @classmethod
     def check_business_eligibility(
-        cls, identifier: str, exclude_in_dissolution=True
+        cls, identifier: str, exclude_settings: ExclusionSettings = ExclusionSettings()
     ) -> Tuple[bool, EligibilityDetails]:
         """Return true if the business with provided identifier is eligible for dissolution.
 
         Returns:
             eligible (bool): True if the business is eligible for dissolution.
             eligibility_details (EligibilityDetails): Details regarding eligibility.
         """
-        query = cls._get_businesses_eligible_query(exclude_in_dissolution).\
-            filter(Business.identifier == identifier)
+        query = cls._get_businesses_eligible_query(
+            exclude_settings.exclude_in_dissolution, exclude_settings.exclude_future_effective_filing
+        ).filter(Business.identifier == identifier)
```

Comment:
can we just pass the new dataclass(`EligibilityFilters`) for all functions?

Reply:
Updated